op-gateway/src/server.ts

```diff
@@ -21,7 +26,12 @@ const program = new Command()
     'address for L2 output oracle on the L1',
     ''
   )
-  .option('-d, --delay <number>', 'number of blocks delay to use', '5');
+  .option(
+    '-d, --delay <number>',
+    'number of delay to use (in minutes for DisputeGame)',
+    '5'
+  )
+  .option('-t, --type <number>', '0 - L2OutputOracle, 1 - DisputeGame', '1');
```

Comment:
`version`, perhaps? Or better yet accept two different arguments, to supply either the L2OutputOracle or DisputeGameFactory and choose the implementation on that basis.

Reply:
I'm thinking to have a separate path prefix like `/v1` for L2OutputOracle, `/v2` for DisputeGameFactory and `/vN` for future upgrades. This would allow old immutable OPVerifier contract to continue working after the gateway upgrade.

Reply:
Do OP expect to continue to operate the old contracts on the same network once the new ones are deployed?

Reply:
We can do it in a simpler way by passing OptimismPortalProxy address instead of either L2OutputOracle or DisputeGameFactory
* OptimismPortal can resolve to l2Oracle only on the old contract
* OptimismPortal can resolve to disputeGameFactory only on the new contract (Because old method is replaced by spacer_54_0_20)

This make it possible to automatically determine whether OP is using an new or old implementation